tests/basic/async_dr_test.go

```diff
@@ -376,3 +690,135 @@ func DeleteAndWaitForMigrationDeletion(name, namespace string) error {
 	_, err = task.DoRetryWithTimeout(getMigration, migrationRetryTimeout, migrationRetryInterval)
 	return err
 }
+
+func initialSetupApps(taskNamePrefix string, single bool) ([]string, []*scheduler.Context) {
+	var contexts []*scheduler.Context
+	var migrationNamespaces []string
+
+	err = SetCustomKubeConfig(asyncdr.FirstCluster)
+	log.FailOnError(err, "Switching context to first cluster failed")
+	if single {
+		taskName := fmt.Sprintf("%s", taskNamePrefix)
+		log.Infof("Task name %s\n", taskName)
+		contexts = append(contexts, ScheduleApplications(taskName)...)
+	} else {
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			log.Infof("Task name %s\n", taskName)
+			contexts = append(contexts, ScheduleApplications(taskName)...)
+		}
+	}
+	for _, ctx := range contexts {
+		// Override default App readiness time out of 5 mins with 10 mins
+		ctx.ReadinessTimeout = appReadinessTimeout
+		namespace := GetAppNamespace(ctx, "")
+		migrationNamespaces = append(migrationNamespaces, namespace)
+	}
+	log.Infof("Migration Namespaces are : [%v]", migrationNamespaces)
+	ValidateApplications(contexts)
+	return migrationNamespaces, contexts
+}
+
+func createMigSchdAndValidateMigration(migSchedName, cpName, migNs, resetConfigPath string, extraArgs map[string]string) {
+	var migration *storkapi.Migration
+	err = storkctlcli.ScheduleStorkctlMigrationSched(migSchedName, cpName, migNs, extraArgs)
+	log.FailOnError(err, "Error creating migrationschedule: %v", err)
+	err = hardSetConfig(resetConfigPath)
+	log.FailOnError(err, "Error setting destination config: %v", err)
+	migSchedule, err := storkops.Instance().GetMigrationSchedule(migSchedName, migNs)
+	log.FailOnError(err, "failed to get migrationschedule %v, err: %v", migSchedName, err)
+	time.Sleep(time.Second * 30)
+	migrations := migSchedule.Status.Items["Interval"]
+	for _, mig := range migrations {
+		migration, err = storkops.Instance().GetMigration(mig.Name, migNs)
+		log.FailOnError(err, "failed to get migration for migrationschedule %v, err: %v", migSchedName, err)
+		err = WaitForMigration([]*storkapi.Migration{migration})
+		log.FailOnError(err, "Migration failed with error: %v", err)
+	}
+}
+
+func performFailoverFailback(foFbParams failoverFailbackParam) {
+	err, output := storkctlcli.PerformFailoverOrFailbackStlCli(foFbParams.action, foFbParams.failoverOrFailbackNs, foFbParams.migrationSchedName, foFbParams.skipSourceOp, foFbParams.extraArgsFailoverFailback)
```

Comment:
nit: `PerformFailoverOrFailbackStlCli` ==> `PerformFailoverOrFailback` since it's from the `storkctlcli` package already.

Reply:
Done.